File: Ahorn/triggers/holdableDashTrigger.jl
module SSMHelperHoldableDashTrigger

using ..Ahorn, Maple

@mapdef Trigger "SSMHelper/HoldableDashTrigger" HoldableDashTrigger(x::Integer, y::Integer, width::Integer=16, height::Integer=16,
    mode::String="EnableOnStay")

const modes = String["EnableOnStay", "DisableOnStay", "EnableToggle", "DisableToggle"]

const placements = Ahorn.PlacementDict(
    "Dash With Holdable Trigger (SSM Helper)" => Ahorn.EntityPlacement(
        HoldableDashTrigger,
        "rectangle"
    )
)

Ahorn.editingOptions(entity::HoldableDashTrigger) = Dict{String, Any}(
    "mode" => modes
)

end

File: Loenn/triggers/holdableDashTrigger.lua
local holdableDashTrigger = {}

local modes = {"EnableOnStay", "DisableOnStay", "EnableToggle", "DisableToggle"}

holdableDashTrigger.name = "SSMHelper/HoldableDashTrigger"
holdableDashTrigger.placements = {
    name = "normal",
    data = {
        mode = "EnableOnStay"
    }
}

holdableDashTrigger.fieldInformation = {
    mode = {
        options = modes,
        editable = false
    }
}

return holdableDashTrigger
